pkg/driver/node.go

```diff
@@ -848,29 +839,104 @@ func collectMountOptions(fsType string, mntFlags []string) []string {
 	return options
 }
 
+// startNotReadyTaintWatcher launches a short‑lived Node informer that removes the
+// ebs.csi.aws.com/agent‑not‑ready taint as soon as both the taint is present and
+// the driver’s Allocatable value is set. The watcher stops automatically after
+// maxWatchDuration or immediately after successful removal.
+func startNotReadyTaintWatcher(clientset kubernetes.Interface, maxWatchDuration time.Duration) {
+	nodeName := os.Getenv("CSI_NODE_NAME")
+	if nodeName == "" {
+		klog.V(4).InfoS("CSI_NODE_NAME missing, skipping taint watcher")
+		return
+	}
+
+	stopCh := make(chan struct{})
+	var closeOnce sync.Once
+	closeWatch := func() { closeOnce.Do(func() { close(stopCh) }) }
+
+	factory := informers.NewSharedInformerFactoryWithOptions(
+		clientset,
+		0,
+		informers.WithTweakListOptions(func(lo *metav1.ListOptions) {
+			lo.FieldSelector = "metadata.name=" + nodeName
+		}),
+	)
+	informer := factory.Core().V1().Nodes().Informer()
+
+	attemptTaintRemoval := func(n *corev1.Node) {
+		if !hasNotReadyTaint(n) {
+			return
+		}
+		if err := checkAllocatable(clientset, nodeName); err != nil {
+			return
+		}
+		if err := removeNotReadyTaint(clientset); err != nil {
+			klog.ErrorS(err, "failed to remove agent‑not‑ready taint")
+			return
+		}
+		klog.V(2).InfoS("Successfully removed agent‑not‑ready taint; stopping watcher")
+		closeWatch()
+	}
+
+	if _, err := informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			if n, ok := obj.(*corev1.Node); ok {
+				attemptTaintRemoval(n)
+			}
+		},
+		UpdateFunc: func(_, newObj interface{}) {
+			if n, ok := newObj.(*corev1.Node); ok {
+				attemptTaintRemoval(n)
+			}
+		},
+	}); err != nil {
+		klog.ErrorS(err, "taint‑watcher: failed to add event handler")
```

Comment:
np: start log message with a capital letter (failed -> Failed)

Same np several times in this file later